go-controller/pkg/ovn/gress_policy.go

```diff
@@ -186,28 +180,24 @@ func (gp *gressPolicy) sizeOfAddressSet() int {
 	return gp.peerV4AddressSets.Len() + gp.peerV6AddressSets.Len()
 }
 
-func (gp *gressPolicy) getMatchFromIPBlock(lportMatch, l4Match string) string {
-	var match string
+func (gp *gressPolicy) getMatchFromIPBlock(lportMatch, l4Match string) []string {
+	var ipBlockMatches []string
 	if gp.policyType == knet.PolicyTypeIngress {
-		ipBlockMatch := constructIPStringForACL("src", gp.ipBlockCidr)
-		if l4Match == noneMatch {
-			match = fmt.Sprintf("match=\"%s && %s\"",
-				ipBlockMatch, lportMatch)
-		} else {
-			match = fmt.Sprintf("match=\"%s && %s && %s\"",
-				ipBlockMatch, l4Match, lportMatch)
-		}
+		ipBlockMatches = constructIPBlockStringsForACL("src", gp.ipBlock)
 	} else {
-		ipBlockMatch := constructIPStringForACL("dst", gp.ipBlockCidr)
+		ipBlockMatches = constructIPBlockStringsForACL("dst", gp.ipBlock)
+	}
+	for i, ipBlockMatch := range ipBlockMatches {
```

Comment:
I am curious as to why we are not doing this for loop inside the constructIPBlockStringsForACL itself.

Reply:
done